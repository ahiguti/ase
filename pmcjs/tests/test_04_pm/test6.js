
Perl.Use("FileHandle");
var fh = new Perl("FileHandle", "sampleutf.txt", "r");
var str_utf;
var buf = Perl.NewScalar();
while (true) {
  var n = fh.read(buf, 1024);
  if (!n) {
    break;
  }
  str_utf += buf.GetBinary();
}
fh.close();
str_utf = Perl.BinToText(str_utf);
var buf = Perl.NewScalar();
buf.Set(str_utf);
Perl.Use("Encode");
var encode = Perl.Functions["Encode"].encode;
var str_euc = encode("euc-jp", str_utf);
dump(str_euc);

