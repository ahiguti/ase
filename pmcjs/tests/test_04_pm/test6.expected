75 6e 64 65 66 69 6e 65 64 2d 2d 2d 2d 2d 2d 2d 2d 2d 2d 2d 2d 2d 2d 2d 2d 2d 2d 2d 2d 2d 2d 2d 2d 2d 2d 2d 2d 2d 2d 2d 2d 2d 2d 2d 2d 2d 2d 2d 2d 2d 2d 2d 2d 2d 2d 2d 2d 2d 2d 2d 2d 2d 2d 2d 2d 2d 2d 2d 2d 2d 2d 2d 2d 2d 2d 2d 2d 2d 2d 2d 2d 0a b0 fa bf f4 a4 ce bc f5 a4 b1 c5 cf a4 b7 0a 0a 4a 61 76 61 53 63 72 69 70 74 a4 c8 50 65 72 6c a4 f2 b8 d9 a4 a4 a4 c0 a5 e1 a5 bd a5 c3 a5 c9 b8 c6 a4 d3 bd d0 a4 b7 a4 c7 a4 cf a1 a2 c5 cf a4 b5 a4 ec a4 bf b0 fa bf f4 a4 c8 ca d6 c3 cd a4 ac a4 bd a4 ec a4 be 0a a4 ec a4 ce a5 a4 a5 f3 a5 bf a5 d7 a5 ea a5 bf a4 c7 b0 b7 a4 a8 a4 eb a5 c7 a1 bc a5 bf a4 cb bc ab c6 b0 c5 aa a4 cb c1 ea b8 df a4 cb ca d1 b4 b9 a4 b5 a4 ec a4 de a4 b9 a1 a3 0a 0a a1 a6 4a 61 76 61 53 63 72 69 70 74 a4 ce bf f4 c3 cd 28 c0 b0 bf f4 a4 e4 c9 e2 c6 b0 be ae bf f4 c5 c0 bf f4 29 a4 c8 50 65 72 6c a4 ce bf f4 c3 cd a4 cf a1 a2 a4 ca a4 eb a4 d9 a4 af c3 cd a4 f2 c2 bb a4 ca a4 ef 0a 20 20 a4 ca a4 a4 a4 e8 a4 a6 a4 cb ca d1 b4 b9 a4 b5 a4 ec a4 de a4 b9 a1 a3 4a 61 76 61 53 63 72 69 70 74 a4 c7 b0 b7 a4 a8 a4 eb c0 b0 bf f4 a4 ce c8 cf b0 cf a4 cf 50 65 72 6c a4 ce a4 bd a4 ec a4 e8 a4 ea a4 e2 0a 20 20 b6 b9 a4 a4 a4 bf a4 e1 a1 a2 50 65 72 6c a4 ab a4 e9 4a 61 76 61 53 63 72 69 70 74 a4 d8 a4 ce bc f5 a4 b1 c5 cf a4 b7 bb fe a4 cb c0 b0 bf f4 a4 ab a4 e9 c9 e2 c6 b0 be ae bf f4 c5 c0 bf f4 a4 cb b3 ca be e5 0a 20 20 a4 b2 a4 b5 a4 ec a4 eb a4 b3 a4 c8 a4 ac a4 a2 a4 ea a4 de a4 b9 a1 a3 0a a1 a6 4a 61 76 61 53 63 72 69 70 74 a4 ce ca b8 bb fa ce f3 28 55 6e 69 63 6f 64 65 ca b8 bb fa ce f3 29 a4 cf a1 a2 55 54 46 31 36 a4 ab a4 e9 55 54 46 38 a4 d8 a4 ce ca d1 b4 b9 a4 cb a4 e8 a4 c3 a4 c6 50 65 72 6c 0a 20 20 a4 ce ca b8 bb fa ce f3 a4 cb ca d1 b4 b9 a4 b5 a4 ec a4 de a4 b9 a1 a3 ca d1 b4 b9 b7 eb b2 cc a4 ce 70 65 72 6c a5 b9 a5 ab a5 e9 a4 cb a4 cf 75 74 66 38 a5 d5 a5 e9 a5 b0 a4 ac c9 d5 a4 ad a4 de a4 b9 a1 a3 0a 20 20 a4 e2 a4 b7 4a 61 76 61 53 63 72 69 70 74 ca b8 bb fa ce f3 a4 f2 50 65 72 6c a4 ce a5 d0 a5 a4 a5 c8 ce f3 28 a5 d0 a5 a4 a5 ca a5 ea a5 c7 a1 bc a5 bf 29 a4 d8 a4 c8 ca d1 b4 b9 a4 b7 a4 bf a4 a4 be ec b9 e7 0a 20 20 a4 cf a1 a2 0a 20 20 20 20 76 61 72 20 78 20 3d 20 50 65 72 6c 2e 4e 65 77 53 63 61 6c 61 72 28 29 3b 0a 20 20 20 20 78 2e 53 65 74 42 69 6e 61 72 79 28 73 74 72 29 3b 0a 20 20 a4 c8 a4 b7 a4 c6 a1 a2 73 74 72 a4 ce a4 ab a4 ef a4 ea a4 cb 78 a4 f2 c5 cf a4 b7 a4 c6 a4 e4 a4 ea a4 de a4 b9 a1 a3 53 65 74 42 69 6e 61 72 79 28 29 a4 cf a1 a2 b0 fa bf f4 a4 cb bb d8 c4 ea a4 b5 a4 ec 0a 20 20 a4 bf ca b8 bb fa ce f3 28 55 6e 69 63 6f 64 65 ca b8 bb fa ce f3 29 a4 ce b3 c6 ca b8 bb fa 28 55 43 53 32 ca b8 bb fa 29 a4 ce b2 bc b0 cc 38 62 69 74 a4 c0 a4 b1 a4 f2 bc e8 a4 ea bd d0 a4 b7 a4 c6 c6 c0 0a 20 20 a4 e9 a4 ec a4 eb a5 d0 a5 a4 a5 c8 ce f3 a4 f2 70 65 72 6c a4 ce a5 b9 a5 ab a5 e9 a4 cb a5 bb a5 c3 a5 c8 a4 b7 a4 de a4 b9 a1 a3 a4 bd a4 ce b7 eb b2 cc c6 c0 a4 e9 a4 ec a4 bf a5 b9 a5 ab a5 e9 a4 cb a4 cf 0a 20 20 75 74 66 38 a5 d5 a5 e9 a5 b0 a4 cf c9 d5 a4 ad a4 de a4 bb a4 f3 a1 a3 0a a1 a6 50 65 72 6c a4 ce ca b8 bb fa ce f3 a4 cf a1 a2 a4 bd a4 ce ca b8 bb fa ce f3 a4 cb 75 74 66 38 a5 d5 a5 e9 a5 b0 a4 ac c9 d5 a4 a4 a4 c6 a4 a4 a4 eb be ec b9 e7 a4 cb a4 cf 55 54 46 38 a4 ab a4 e9 0a 20 20 55 54 46 31 36 a4 d8 a4 ce ca d1 b4 b9 a4 cb a4 e8 a4 c3 a4 c6 4a 61 76 61 53 63 72 69 70 74 ca b8 bb fa ce f3 a4 cb ca d1 b4 b9 a4 b5 a4 ec a4 de a4 b9 a1 a3 75 74 66 38 a5 d5 a5 e9 a5 b0 a4 ac c9 d5 a4 a4 0a 20 20 a4 c6 a4 a4 a4 ca a4 a4 be ec b9 e7 a4 cb a4 cf a1 a2 a4 bd a4 ce a5 b9 a5 ab a5 e9 c3 cd a4 ce b3 c6 a5 d0 a5 a4 a5 c8 a4 ce c3 cd a4 f2 a4 bd a4 ce a4 de a4 de 55 6e 69 63 6f 64 65 a5 b3 a1 bc a5 c9 a5 dd 0a 20 20 a5 a4 a5 f3 a5 c8 a4 c8 b8 ab d0 f6 a4 b7 a4 bf 55 6e 69 63 6f 64 65 ca b8 bb fa ce f3 a4 d8 a4 c8 ca d1 b4 b9 a4 b5 a4 ec a4 de a4 b9 a1 a3 a4 c4 a4 de a4 ea a1 a2 a5 c6 a5 ad a5 b9 a5 c8 a5 c7 a1 bc a5 bf 0a 20 20 28 75 74 66 38 a5 d5 a5 e9 a5 b0 c9 d5 a4 ad 29 a4 cf c1 b0 bc d4 a4 ce ca d1 b4 b9 a1 a2 a5 d0 a5 a4 a5 ca a5 ea a5 c7 a1 bc a5 bf 28 75 74 66 38 a5 d5 a5 e9 a5 b0 cc b5 a4 b7 29 a4 cf b8 e5 bc d4 a4 ce 0a 20 20 ca d1 b4 b9 a4 ac c5 ac cd d1 a4 b5 a4 ec a4 de a4 b9 a1 a3 0a a1 a6 50 65 72 6c a4 ce bb b2 be c8 b7 bf a4 ce c3 cd a4 cf a1 a2 4a 61 76 61 53 63 72 69 70 74 a4 ce a5 aa a5 d6 a5 b8 a5 a7 a5 af a5 c8 28 a5 aa a5 d6 a5 b8 a5 a7 a5 af a5 c8 bb b2 be c8 29 a4 d8 a4 c8 ca d1 b4 b9 0a 20 20 a4 b5 a4 ec a4 de a4 b9 a1 a3 a4 e2 a4 b7 a4 bd a4 ce bb b2 be c8 b7 bf a4 ce c3 cd a4 ac 50 65 72 6c a4 ce a5 aa a5 d6 a5 b8 a5 a7 a5 af a5 c8 28 62 6c 65 73 73 a4 b5 a4 ec a4 bf bb b2 be c8 29 a4 ca a4 e9 0a 20 20 a4 d0 a1 a2 ca d1 b4 b9 a4 b5 a4 ec a4 bf a5 aa a5 d6 a5 b8 a5 a7 a5 af a5 c8 bb b2 be c8 a4 cb c2 d0 a4 b7 a4 c6 20 6f 62 6a 2e 6d 65 74 68 6f 64 28 61 72 67 73 2e 2e 2e 29 20 a4 ce b7 c1 bc b0 a4 c7 a5 e1 0a 20 20 a5 bd a5 c3 a5 c9 a4 f2 b8 c6 a4 d3 bd d0 a4 b9 a4 b3 a4 c8 a4 ac a4 c7 a4 ad a4 de a4 b9 a1 a3 bb b2 be c8 c0 e8 a4 ac 50 65 72 6c a4 ce a5 cf a5 c3 a5 b7 a5 e5 a5 c6 a1 bc a5 d6 a5 eb a4 e4 c7 db ce f3 a4 ce 0a 20 20 be ec b9 e7 a4 cf a1 a2 6f 62 6a 2e 70 72 6f 70 65 72 74 79 20 a4 e4 20 6f 62 6a 5b 22 70 72 6f 70 65 72 74 79 22 5d 20 a4 ce b7 c1 bc b0 a4 c7 cd d7 c1 c7 a4 f2 c6 c9 a4 df bd f1 a4 ad a4 c7 a4 ad a4 de a4 b9 a1 a3 0a a1 a6 4a 61 76 61 53 63 72 69 70 74 a4 ce a5 aa a5 d6 a5 b8 a5 a7 a5 af a5 c8 bb b2 be c8 a4 cf a1 a2 50 65 72 6c a4 ce a5 aa a5 d6 a5 b8 a5 a7 a5 af a5 c8 bb b2 be c8 28 62 6c 65 73 73 a4 b5 a4 ec a4 bf 0a 20 20 68 61 73 68 72 65 66 29 a4 d8 a4 c8 ca d1 b4 b9 a4 b5 a4 ec a4 de a4 b9 a1 a3 20 24 6f 62 6a 2d 3e 6d 65 74 68 6f 64 28 61 72 67 73 2e 2e 2e 29 20 a4 ce b7 c1 bc b0 a4 c7 a5 e1 a5 bd a5 c3 a5 c9 a4 f2 b8 c6 0a 20 20 a4 d3 bd d0 a4 b7 a4 bf a4 ea a1 a2 24 6f 62 6a 2d 3e 7b 27 70 72 6f 70 65 72 74 79 27 7d 20 a4 ce b7 c1 bc b0 a4 c7 a5 d7 a5 ed a5 d1 a5 c6 a5 a3 a4 f2 c6 c9 a4 df bd f1 a4 ad a4 c7 a4 ad a4 de a4 b9 a1 a3 0a 0a